agent-bridge-datastore/src/main/java/com/newrelic/agent/bridge/datastore/JdbcHelper.java

```diff
@@ -42,27 +42,47 @@ protected Boolean initialValue() {
 
     /**
      * Any values in the urlToFactory or urlToDatabaseName Maps older than this value will
-     * be evicted from the cache when the {@link ExpiringValueCache} timer fires. This
+     * be evicted from the cache when the {@link ExpiringValueMap} timer fires. This
      * will also be used as the timer interval value.
      */
     private static final long CACHE_EXPIRATION_AGE_MILLI = Duration.ofHours(8).toMillis();
 
-    private static final ExpiringValueCache.ExpiringValueLogicFunction EXPIRE_FUNC =
+    private static final ExpiringValueMap.ExpiringValueLogicFunction EXPIRE_FUNC =
             (timeCreated, timeLastAccessed) -> timeLastAccessed < (System.currentTimeMillis() - CACHE_EXPIRATION_AGE_MILLI);
 
     // This will contain every vendor type that we detected on the client system
     private static final Map<String, DatabaseVendor> typeToVendorLookup = new ConcurrentHashMap<>(10);
     private static final Map<Class<?>, DatabaseVendor> classToVendorLookup = AgentBridge.collectionFactory.createConcurrentWeakKeyedMap();
-    private static final ExpiringValueCache<String, ConnectionFactory> urlToFactory =
-            new ExpiringValueCache<>("urlToFactoryCache", CACHE_EXPIRATION_AGE_MILLI, EXPIRE_FUNC);
-
-    private static final ExpiringValueCache<String, String> urlToDatabaseName =
-            new ExpiringValueCache<>("urlToDatabaseNameCache", CACHE_EXPIRATION_AGE_MILLI, EXPIRE_FUNC);
     private static final Map<Statement, String> statementToSql = AgentBridge.collectionFactory.createConcurrentWeakKeyedMap();
     private static final Map<Connection, String> connectionToIdentifier = AgentBridge.collectionFactory.createConcurrentWeakKeyedMap();
     private static final Map<Connection, String> connectionToURL = AgentBridge.collectionFactory.createConcurrentWeakKeyedMap();
     public static final String UNKNOWN = "unknown";
 
+    // Config to toggle the use of the ExpiringValueCache Map implementation or vanilla ConcurrentHashMap implementation
+    // for the urlToFactory and urlToDatabaseName maps. The default is to use the ExpiringValueCache. Set a config
+    // property of use_jdbchelper.vanilla_map=true to revert to the ConcurrentHashMap implementation.
+    private static final Map<String, ConnectionFactory> urlToFactory;
+    private static final Map<String, String> urlToDatabaseName;
+    static {
+        boolean useVanillaMap = false;
+        try {
+            useVanillaMap = NewRelic.getAgent().getConfig().getValue("use_jdbchelper_vanilla_map");
+        } catch (Exception ignored) {
+            // Just in case the getValue call throws an NPE
```

Comment:
You can use the getValue(String key, T defaultValue)
This way it will never return null and never call a NPE.

Reply:
Corrected